src/sql/workbench/services/notebook/browser/sql/sqlSessionManager.ts

```diff
@@ -620,6 +624,7 @@ export class SQLFuture extends Disposable implements FutureInternal {
 
 	private convertToError(msg: IResultMessage | string): nb.IIOPubMessage {
 		if (msg) {
+			this._errorOccurred = true;
```

Comment:
Does a new session get created when the query is re-ran? If not shouldn't you be clearing this somewhere?

Reply:
A new future does when the query is re-ran. Made sure to test that scenario.